File: Producto/routes-service/src/main/java/com/tagok/routes_service/domain/dto/response/PorticoResponse.java
package com.tagok.routes_service.domain.dto.response;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PorticoResponse 
{
    private Long id;
    private String codigo;
    private String nombre;
    private String sentido;
    private double latitud;
    private double longitud;

    private String autopista;

    private List<ReglaTarifariaResponse> reglas;
    private CalendarioTarifarioResponse calendario;
}

File: Producto/tag-ok-app/app/src/main/java/com/tagok/app/data/dto/portico/PorticoResponse.kt
package com.tagok.app.data.dto.portico

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName("PORTICO")
data class PorticoResponse(
    val id: Long,
    val codigo: String? = null,
    val nombre: String? = null,
    val sentido: String? = null,
    val latitud: Double,
    val longitud: Double,
    val autopista: String,
    val reglas: List<ReglaTarifariaResponse>,
    val calendario: CalendarioTarifarioResponse) : TollResponse()
